### PR TITLE
Admin

### PR DESCRIPTION
**Pull request by lunore**

### CODE REVIEW

File: src/components/ManageUsers/ManageUsers.jsx [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 const ManageUsers = () => {
   return (
     <Layout>
-      <p>Manage Users</p>
+      <UserSearch />
+      <ul className="manage-users_list">
+        {users.map((user) => (
+          <UsersItem key={user.email} user={user} />
+        ))}
+      </ul>
```
Comment (KubaC701) [inline src/components/ManageUsers/ManageUsers.jsx:15]: nice component splitting

File: src/components/ManageUsers/UserItem.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      return 'admin.png';
+    } else if (user.role === 'librarian') {
+      return 'librarian.png';
+    } else {
+      return 'customer.png';
+    }
+  };
+  const icon = getIcon();
+  return (
+    <li className="useritem">
```
Comment (KubaC701) [inline src/components/ManageUsers/UserItem.jsx]: 2-word classes should be separated by `-`. so in this case, it should be
```suggestion
    <li className="user-item">
```
Please change it in all places

File: src/components/ManageUsers/UserSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const UserSearch = () => {
+  return (
+    <main>
+      <div className="usersearch__container">
+        <div>
+          <h1 className="usersearch__h1">Enter user email:</h1>
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: it is not a h1. It's a label. See the usage in the login form here: https://github.com/KubaC701/library-webiste/blob/main/src/components/login/Login.jsx#LL38C17-L38C17

File: src/components/ManageUsers/UserSearch.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        <div className="usersearch__search">
+          <input
+            className="usersearch__input"
+            type="text"
+            name="search"
+            id="search"
+            placeholder="example@lib.com"
+          />
+        </div>
+        <button type="active" className="usersearch__adduserbutton">
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: type="active" is not a valid type. I believe you wanted to use:
```
        <button type="button" className="usersearch__adduserbutton">
```
or even better:
```
        <button type="submit" className="usersearch__adduserbutton">
```
but then you need to wrap the input and the button in the form tag. It allows submit the form by clicking enter

File: src/components/ManageUsers/manage-users.css [hunk 5]
```diff
@@ -0,0 +1,84 @@
+.usersearch__h1 {
+  font-size: 16px;
+  color: #e3893e;
+  display: right;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: there is no `display:right`. If it works it is to remove

File: src/components/ManageUsers/manage-users.css [hunk 6]
```diff
@@ -0,0 +1,84 @@
+.usersearch__h1 {
+  font-size: 16px;
+  color: #e3893e;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]:
```suggestion
  color: var(--color-primary);
```

Change everywhere where you are using our orange color

File: src/components/ManageUsers/manage-users.css [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  display: right;
+  font-weight: 200px;
+  margin-right: 10px;
+  text-align: right;
+}
+
+.usersearch__container {
+  padding: 16px;
+  min-height: 10vh;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: don't use `min-height` in that case. The height will be set automatically

File: src/components/ManageUsers/manage-users.css [hunk 8]
```diff
@@ -0,0 +1,84 @@
+.usersearch__h1 {
+  font-size: 16px;
+  color: #e3893e;
+  display: right;
+  font-weight: 200px;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: `font-weight` has no pixels. On figma the text is in semi-bold so the corresponding value is 600
```suggestion
  font-weight: 600;
```

File: src/components/ManageUsers/manage-users.css [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.usersearch__container {
+  padding: 16px;
+  min-height: 10vh;
+  background: #000000;
+  background-size: cover;
+  display: flex;
+  flex-direction: column;
+  margin-left: auto;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: all those values are unnecessary

File: src/components/ManageUsers/manage-users.css [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.usersearch__search {
+  margin-top: 10px;
+  margin-bottom: 10px;
+  gap: 10px;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: gap works only with flex. It has no effect here

File: src/components/ManageUsers/manage-users.css [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  background: #000000;
+  background-size: cover;
+  display: flex;
+  flex-direction: column;
+  margin-left: auto;
+}
+
+.usersearch__search {
+  margin-top: 10px;
+  margin-bottom: 10px;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: it can be shortened to:
```suggestion
  margin: 10px 0;
```

File: src/components/ManageUsers/manage-users.css [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  background-color: #242424;
+  border-radius: 8px;
+  list-style: none;
+}
+
+.useritem {
+  padding: 4px 0;
+  color: white;
+  margin-right: auto;
+  margin-top: auto;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: to remove

File: src/components/UsersHistory/UsersHistory.jsx
Comment (KubaC701) [inline src/components/UsersHistory/UsersHistory.jsx:1]: why this file is here?

File: src/components/ManageUsers/UserSearch.jsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+const UserSearch = () => {
+  return (
+    <main>
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: main tag here is not the best option. We already have a container which is main. You can remove it and leave only `usersearch__container` div

**Review by KubaC701 — CHANGES_REQUESTED** [review]
Try to do it as it is in the Figma. It looks better
image

File: src/components/ManageUsers/UserSearch.jsx (hunk 13 above)
Comment (lunore) [reply src/components/ManageUsers/UserSearch.jsx]: Main tag was removed as you suggested.

File: src/components/ManageUsers/UserItem.jsx (hunk 2 above)
Comment (lunore) [reply src/components/ManageUsers/UserItem.jsx]: name class "useritem" was changed to "user-item" in each recurence.

File: src/components/ManageUsers/UserSearch.jsx (hunk 3 above)
Comment (lunore) [reply src/components/ManageUsers/UserSearch.jsx]: The "tag" of element  was changed to <label classname="usersearch_label"> ...etc . Class name was changed in each recurrence.

File: src/components/ManageUsers/manage-users.css [hunk 14]
```diff
@@ -1,58 +1,50 @@
...
-  max-width: 300px;
-  margin-right: 10px;
-  margin-left: auto;
+  max-width: 500px;
 }
 
 .usersearch__input {
   height: 16px;
   width: 100%;
-  max-width: 300px;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: why did you add 500px width to the container (usersearch__search) and here 600? It's enough to put it in one place

File: src/components/UsersHistory/UsersHistory.jsx
Comment (lunore) [reply src/components/UsersHistory/UsersHistory.jsx:1]: IDK....

Comment (lunore) [reply src/components/UsersHistory/UsersHistory.jsx:1]: It's not my code and I didn't add this on my branch.

File: src/components/ManageUsers/UserSearch.jsx (hunk 4 above)
Comment (lunore) [reply src/components/ManageUsers/UserSearch.jsx]: Button type was changed to "button" (masło maślane)

File: src/components/ManageUsers/manage-users.css (hunk 6 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Has done for manage-users.css

File: src/components/ManageUsers/manage-users.css (hunk 5 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Already deleted

File: src/components/ManageUsers/manage-users.css (hunk 7 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Removed

File: src/components/ManageUsers/manage-users.css (hunk 8 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Already changed that in latest commit

File: src/components/ManageUsers/manage-users.css (hunk 9 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Disagree. "display: flex" and "flex-direction: column" should stay in my opinion.

Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: The rest was removed.

File: src/components/ManageUsers/manage-users.css (hunk 11 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Fixed

File: src/components/ManageUsers/manage-users.css (hunk 14 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: sorry, I don't see it.... maybe I have fixed this in last commit...

File: src/components/ManageUsers/manage-users.css (hunk 10 above)
Comment (lunore) [reply src/components/ManageUsers/manage-users.css]: Has been removed in last commit

File: src/components/ManageUsers/UserItem.jsx [hunk 15]
```diff
@@ -10,10 +10,10 @@ function UserItem({ user }) {
   };
   const icon = getIcon();
   return (
-    <li className="useritem">
+    <li className="user-item">
       <img src={`/icons/${icon}`} />
-      <p className="useritem__email">{user.email}</p>
-      <button type="active" className="useritem_delete">
+      <p className="user-item__email">{user.email}</p>
+      <button type="submit" className="user-item_delete">
```
Comment (KubaC701) [inline src/components/ManageUsers/UserItem.jsx]: this is button type="button" because it fires some action and has no form there https://html.com/attributes/button-type/

File: src/components/ManageUsers/UserSearch.jsx [hunk 16]
```diff
@@ -0,0 +1,23 @@
+const UserSearch = () => {
+  return (
+    <div className="usersearch__container">
+      <div>
+        <label className="usersearch__label">Enter user email:</label>
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: label must have a `for` attribute corresponding with input id, so in your case it will be:
```suggestion
        <label className="usersearch__label" for="search">Enter user email:</label>
     
```
and I guess the div around the label is redundant

File: src/components/ManageUsers/UserSearch.jsx [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+      <div className="usersearch__search">
+        <input
+          className="usersearch__input"
+          type="text"
+          name="search"
+          id="search"
+          placeholder="example@lib.com"
+        />
+      </div>
+      <button type="active" className="usersearch__adduserbutton">
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: I made a mistake here. It is an independent button so it should be also type="button"

File: src/components/ManageUsers/UserSearch.jsx [hunk 18]
```diff
@@ -0,0 +1,23 @@
+const UserSearch = () => {
+  return (
+    <div className="usersearch__container">
```
Comment (KubaC701) [inline src/components/ManageUsers/UserSearch.jsx]: here the same with classes
```suggestion
    <div className="user-search__container">
```

File: src/components/ManageUsers/manage-users.css [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+  width: 100%;
+  max-width: 600px;
+}
+
+.usersearch__input {
+  height: 16px;
+  width: 100%;
+  max-width: 600px;
+  background-color: #242424;
+  color: #000000;
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: I think the font color should be white. 
```suggestion
  color: var(--color-text);
```

**Review by KubaC701 — CHANGES_REQUESTED** [review]

File: src/components/UsersHistory/UsersHistory.jsx
Comment (KubaC701) [reply src/components/UsersHistory/UsersHistory.jsx:1]: we need to remove it. Just delete the `UsersHistory` folder

File: src/components/ManageUsers/manage-users.css [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+  width: 100%;
+  max-width: 600px;
+  background-color: #242424;
+  color: var(--color-text);
+  border-radius: 8px;
+  padding: 20px;
+  border: none;
+}
+
+.user-search__adduserbutton {
```
Comment (KubaC701) [inline src/components/ManageUsers/manage-users.css]: here the same rule applies
```suggestion
.user-search__add-user-button {
```

**Review by KubaC701 — APPROVED** [review]
